childhood/vulnerability/ive-junaeb.ipynb

```diff
@@ -0,0 +1,683 @@
+    "        df[tb] = df[tb].drop(df[tb].columns[14:], axis=1)\n",
+    "        last_row = df[tb].shape[0]\n",
+    "        df[tb] = df[tb].drop(last_row-1)\n",
+    "\n",
+    "        # fills NaN with 0\n",
+    "        df[tb].iloc[:, 9:] = df[tb].iloc[:, 9:].fillna(0)\n",
+    "\n",
+    "        # creates new column to make dataframe tidy\n",
+    "        if tb == 'BÁSICA':\n",
+    "            df[tb][\"level\"] = pd.Series(['BASICA'] * df[tb].shape[0])\n",
+    "        else:\n",
+    "            df[tb][\"level\"] = pd.Series([tb] * df[tb].shape[0])\n",
+    "            \n",
+    "        # creates new year column\n",
+    "        df[tb][\"year\"] = pd.Series([year] * df[tb].shape[0])\n",
+    "\n",
+    "        # melts dataframe to make it tidy\n",
+    "        priorities = df[tb].columns[9:14]\n",
+    "        df[tb] = pd.melt(df[tb], id_vars = [x for x in df[tb].columns if x not in priorities], value_vars = priorities, var_name = \"priority\", value_name = \"total\")\n",
+    "\n",
+    "        # changes column names\n",
+    "        df[tb].columns = [\"rbd\", \"dv_rbd\", \"school_name\", \"dependency\", \"area\", \"region_code\", \"province_code\", \"commune_code\", \"commune_name\", \"level\", \"year\", \"priority\", \"total\"]\n",
+    "\n",
+    "        # changes column types\n",
+    "        num_cols = [\"rbd\", \"dv_rbd\", \"region_code\", \"province_code\", \"commune_code\", \"total\"]\n",
+    "        df[tb][num_cols] = df[tb][num_cols].apply(pd.to_numeric, downcast=\"integer\")\n",
+    "        \n",
+    "        # changes string columns to uppercase\n",
+    "        def uppercase(row, col):\n",
+    "            return row[col].upper()\n",
+    "        \n",
+    "        str_cols = [x for x in df[tb].columns if x not in num_cols]\n",
+    "        \n",
+    "        for column in str_cols:\n",
+    "            df[tb][column] = df[tb].apply(uppercase, col=column, axis=1) \n",
+    "         \n",
+    "    # concatenates both dataframes\n",
+    "    df = pd.concat([df[sheets[0]], df[sheets[1]]], ignore_index=True)\n",
+    "    years_df.append(df)\n",
+    "\n",
+    "# function to get year information\n",
+    "def get_year(string):\n",
+    "    reg = re.search(\"\\d\", string)\n",
+    "    first = reg.start()\n",
+    "    y = string[first : first + 4]\n",
+    "    return y \n",
+    "\n",
+    "# processes each link\n",
+    "for file in os.listdir():\n",
+    "    new_filename = file[file.rfind(\"/\") + 1:]\n",
+    "    print(\"{:.2f} s | Cleaning {}\".format(time.perf_counter()-before, new_filename))\n",
+    "    clean(new_filename, get_year(new_filename))\n",
+    "\n",
+    "# concatenates each year's dataframe\n",
+    "df = pd.concat(years_df, ignore_index=True)\n",
+    "print(\"{:.2f} s | Concatenated each year's dataframe.\".format(time.perf_counter()-before))\n",
+    "\n",
+    "# classifies dependencies\n",
+    "def classify_dependencies(row, col):\n",
```

Comment:
maybe a better option is to create:
```
asdf = {
  "SUB": "PARTICULAR SUBVENCIONADO",
  ...
}
```

And after you can use: `return asdf[row[col]]`

Reply:
I followed the advice and created the dictionary.